lxd/identities.go

```diff
@@ -546,6 +566,47 @@ func tlsIdentityTokenValidate(ctx context.Context, s *state.State, token api.Cer
 	return uid, nil
 }
 
+func clusterLinkIdentityTokenValidate(ctx context.Context, s *state.State, token api.CertificateAddToken) (uuid.UUID, error) {
```

Comment:
Discussed in 1:1 - ignore refactoring comments on DB functions from earlier. Create one function which gets either a pending TLS or pending cluster link identity by token secret and have that query return the type of identity.